### DISCUSSION TITLE
Configuration file for sensoHOME VRT 380 Thermostat

### DISCUSSION THREAD

**Discussion by rRobbieIT** [john30/ebusd#1085] [Q&A] [answered]
Hello, 

I have a thermostat **Vaillant VRT 380** and I could not find any configuration file for it.

The string from ebusctl info is the following:
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV0;SW=0312;HW=9103"

I have tried to rename some "similar" files (looking at their model and name), but the fields do not match and I get a series of ERR.

Any suggestion is welcome, although to reverse engineer the protocol is probably out of my reach.

Thank You
RP

**Comment by rRobbieIT** [comment, marked as answer]
Sorry to answer to my own message, I want to share my findings hoping they could be useful to someone else.

I managed to make the thermostat works, although only with very basic functionality:

- I used the configuration file 15.700.csv renamed to 15.ctlv0.csv
- Using a quick python script I could iterate within all the commands in the csv file, to find the ones which were returning something meaningful. All in all I am only using **z1ActualRoomTempDesired** and **z1RoomTemp**.
- I configured ebusd and MQTT with docker
- I subscribed OpenHAB to the associated MQTT topics as follows:

Reading from Thermostat
`ebusd/ctlv0/z1ActualRoomTempDesired`

Writing to Thermostat
`ebusd/ctlv0/z1ActualRoomTempDesired/set`

**Comment by jtbgroup** [comment]
Hi, 

I just opened a topic https://github.com/john30/ebusd/discussions/1463 which seems very close to yours. 
I can set a lot of parameters, but not the z1ActualRoomTempDesired. Does it still work for you?

Regards

**Comment by rRobbieIT** [comment]
Hi there,

I answered in your topic [https://github.com/john30/ebusd/discussions/1463](url)

**Comment by legacycode** [comment]
Hi everyone,

I have a Vaillant controller, internally identified as sensoHOME VRT380, which appears on the eBUS as CTLV0 (address 15).

My questions:

Which file from the repository is the best starting point for my controller – 15.700.csv, 15.370.csv, 15.ctlv2.tsp, or is there now a complete 15.ctlv0.csv specifically for VRT380(f)?

Is there already a version that covers all registers, or do I need to map them step by step myself?

Thanks in advance for your guidance!

**Reply by d1vzero** [reply]
Hi,
I have Vaillant controller sensoHome VRT 380/2. It appears on the ebus as CTLV3 (address 15).
I experienced good results by using the CTLV2 file. However, not sure if yours is similar. First start only reading values. If the numbers and values you get make sense you can go beyond read-only.

By the way a very good starting point to understand how the differnt registers work together and influence each other is:
https://github.com/stadid/ebusd-configuration/pull/2#issue-2138285601
